controllers/main.go

```diff
@@ -120,14 +120,14 @@ func getClientIP(r *http.Request, realIPHeader string) string {
 }
 
 // NewMainController is the constructor for the entire controller routing.
-func NewMainController(mainController MainController) (*MainController, error) {
+func NewMainController(mainController *MainController) (*MainController, error) {
 
 	ch := &CaptchaHandler{
 		ImgHeight: 127,
 		ImgWidth:  257,
 	}
 
-	mc := &mainController
+	mc := mainController
```

Comment:
I'm sorry, I meant that there's no need to copy anything here, you can adjust the "mainController" and return that.